pkg/secretless/plugin/v1/listener.go

```diff
@@ -47,9 +47,11 @@ type BaseListener struct {
 	Config         config.Listener
 	EventNotifier  EventNotifier
 	HandlerConfigs []config.Handler
+	IsShuttingDown bool
```

Comment:
Perhaps there's a more accurate name for this field e.g. `ShutdownStarted`, which would be true while the listener is shutting down and has already shut down.